BDArmory/Ammo/PooledBullet.cs

```diff
@@ -1424,132 +1424,137 @@ public void BeehiveDetonation()
-                    pBullet.defaultDetonationRange = defaultDetonationRange;
-                    pBullet.fuzeType = sFuze;
-                }
-                else
-                {
-                    pBullet.fuzeType = BulletFuzeTypes.None;
-                    pBullet.sabot = ((subMunitionType.bulletMass * 1000 / (subMunitionType.caliber * subMunitionType.caliber * Mathf.PI / 400 * 19) + 1) * 10) > subMunitionType.caliber * 4;
-                    pBullet.HEType = PooledBulletTypes.Slug;
-                }
-                pBullet.EMP = subMunitionType.EMP;
-                pBullet.nuclear = subMunitionType.nuclear;
-                pBullet.beehive = subMunitionType.beehive;
-                pBullet.subMunitionType = BulletInfo.bullets[subMunitionType.subMunitionType];
-                //pBullet.homing = BulletInfo.homing;
-                pBullet.impulse = subMunitionType.impulse;
-                pBullet.massMod = subMunitionType.massMod;
-                switch (subMunitionType.bulletDragTypeName)
+                BulletInfo sBullet = BulletInfo.bullets[projType];
+                string fuze = sBullet.fuzeType.ToLower();
+
+                BulletFuzeTypes sFuze;
+                switch (fuze)
                 {
-                    case "None":
-                        pBullet.dragType = BulletDragTypes.None;
+                    case "timed":
+                        sFuze = BulletFuzeTypes.Timed;
                         break;
-                    case "AnalyticEstimate":
-                        pBullet.dragType = BulletDragTypes.AnalyticEstimate;
+                    case "proximity":
+                        sFuze = BulletFuzeTypes.Proximity;
                         break;
-                    case "NumericalIntegration":
-                        pBullet.dragType = BulletDragTypes.NumericalIntegration;
+                    case "flak":
+                        sFuze = BulletFuzeTypes.Flak;
+                        break;
+                    case "delay":
+                        sFuze = BulletFuzeTypes.Delay;
+                        break;
+                    case "penetrating":
+                        sFuze = BulletFuzeTypes.Penetrating;
+                        break;
+                    case "impact":
+                        sFuze = BulletFuzeTypes.Impact;
+                        break;
+                    case "none":
+                        sFuze = BulletFuzeTypes.Impact;
                         break;
                     default:
-                        pBullet.dragType = BulletDragTypes.AnalyticEstimate;
+                        sFuze = BulletFuzeTypes.None;
                         break;
                 }
-                pBullet.bullet = BulletInfo.bullets[subMunitionType.name];
-                pBullet.stealResources = stealResources;
-                pBullet.dmgMult = dmgMult;
-                pBullet.isAPSprojectile = isAPSprojectile;
-                pBullet.isSubProjectile = true;
```

Comment:
`isSubProjectile = true;` needs to be included here. It's used for determining whether the bullet should arm immediately or not.